std/typecons.d

```diff
@@ -746,6 +777,64 @@ template Tuple(Specs...)
         }
 
         /**
+         * Iterate fields in tuple as a _range.
+         *
+         * Only available when all field types are exactly the same in
+         * unqualified form. When the field types are all the same but with
+         * different mutability, the element type of the _range is `const`.
+         *
+         * Not available for the empty tuple.
+         *
+         * Returns:
+         *     Slice of the fields inside this tuple; the lifetime of the
+         *     slice must be no longer than the lifetime of the tuple.
+         */
+        static if (Types.length && !is(CommonLvalueType!Types == void))
+        {
+            CommonLvalueType!Types[] range() @system
+            {
+                return (&expand[0])[0 .. Types.length];
+            }
+
+            ///
+            nothrow unittest
```

Comment:
If you put the unittest inside the struct, there will be a new unittest instantiated each time the tuple is instantiated. Jonathan Davis pointed this out in the newsgroup and I think it's something we should avoid.